ote/src/cljs/ote/ui/common.cljs

```diff
@@ -12,58 +12,111 @@
             [reagent.core :as r]
             [clojure.string :as str]
             [reagent.core :as reagent]
+            [re-svg-icons.feather-icons :as feather-icons]
 
             [goog.crypt.Md5]
             [goog.crypt]
 
             [ote.util.text :as text]
             [ote.app.routes :as routes]))
 
+(def ^:private quicklink-urls
+  {:fintraffic        {:url "https://www.fintraffic.fi"                   :langs {:fi "/fi" :sv "/sv" :en "/en"}}
+   :traffic-situation {:url "https://liikennetilanne.fintraffic.fi"       :langs {:fi "/fi" :sv "/sv" :en "/en"}}
+   :feedback-channel  {:url "https://palautevayla.fi/aspa?lang="          :langs {:fi "fi"  :sv "sv"  :en "en"}}
+   :train-departures  {:url "https://junalahdot.fi/junalahdot/main?lang=" :langs {:fi "1"   :sv "2"   :en "3"}}
+   :skynavx           {:url "https://skynavx.fi/#/drone"                  :langs {}}
+   :digitraffic       {:url "https://www.digitraffic.fi"                  :langs {:en "/en/"}}
+   :digitransit       {:url "https://digitransit.fi"                      :langs {:en "/en/"}}
+   :finap             {:url "https://finap.fi/#/"                         :langs {}}})
+
+(def ^:private trusted-urls
+  "List of external trusted URLs which user doesn't need to confirm before navigating."
+  (concat
+    (map (fn [[_ data]] (:url data)) quicklink-urls)  ; Fintraffic properties
+    ["https://www.traficom.fi"]                       ; UI links to various resources hosted on Traficom.fi
+    ))
+
+(defn- trusted-url?
+  [url]
+  (some #(clojure.string/starts-with? url %) trusted-urls))
+
+(defn localized-quicklink-uri [quicklink]
+  (let [current-language    (or (keyword @localization/selected-language) :fi)
+        {:keys [url langs]} (get quicklink-urls quicklink)
+        lang                (get langs current-language "")]
+    (str url lang)))
+
 (def mobile?
   (let [ua (str/lower-case js/window.navigator.userAgent)]
     (boolean (some (partial str/includes? ua) ["android" "iphone" "ipad" "mobile"]))))
 
+(defn- sanitize-url
+  "Autodetects and sanitizes given URL based on whether it's internal or external link. Returns a tuple of sanitized URL
+   and a boolean indicating whether the link is internal or external,
+   ```clojure
+   (sanitize-url \"www.example.com\")
+   ;=> [\"http://www.example.com\" true]
+   ```"
+  [url]
+  (cond
+    ;; URL with protocol, use as is
+    (re-matches #"^(\w+:).*" url)
+    [url true]
+
+    ;; User specified link without protocol (like "www.serviceprovider.fi/foo")
+    (re-matches #"^[^/]+\..*" url)
+    [(str "http://" url) true]
```

Comment:
Should `https://` be used here?

Reply:
Not all websites (still! 😢) provide HTTPS, so that would generate more broken links than the current option - most sites do autoredirect from http to https, thankfully. Since these are just strings we can't really do much about it...

Reply:
OK. Can we accept both "https://" and "http://"?

Reply:
On the other hand, that's beyond the scope of this task of warning the user of external links.